src/tools/rest-backend/service_storage.cpp

```diff
@@ -0,0 +1,574 @@
+			entries.erase (entries.begin () + i);
+			entries.push_back (entry);
+			break;
+		}
+	}
+
+	if (!found)
+	{
+		throw exception::EntryNotFoundException ();
+	}
+
+	KeySet ks;
+	this->_kdb.get (ks, entry.getName ());
+
+	Key k = ks.lookup (entry.getName ());
+	if (!k)
+	{
+		throw kdbrest::exception::EntryNotFoundException ();
+	}
+
+	ks.cut (entry);
+	ks.append (entry);
+	ks.append (entry.getSubkeys ());
+
+	if (this->_kdb.set (ks, entry.getName ()) >= 1)
+		return true;
+	else
+		return false;
+}
+
+/**
+         * @brief Allows for deleting of a database entry. Will delete the
+         * entry iteself as well as all meta data and subkeys (configuration).
+         * 
+         * @param entry A custom Entry object that should be deleted.
+         * 
+         * @return true if the entry was deleted successfully, false otherwise
+         * 
+         * @throw kdbrest::exception::EntryNotFoundException in case the entry
+         * to delete does not exist.
+         */
+bool StorageEngine::deleteEntry (model::Entry & entry)
+{
+	using namespace kdb;
+
+	// register exclusive access
+	boost::unique_lock<boost::shared_mutex> lock (_mutex_entry_cache);
+
+	bool found = false;
+	std::vector<model::Entry> & entries = this->_entryCache;
+	for (unsigned int i = 0; i < entries.size (); i++)
+	{
+		if (entries[i].getPublicName ().compare (entry.getPublicName ()) == 0)
+		{
+			found = true;
+			entries.erase (entries.begin () + i);
+			break;
+		}
+	}
+
+	if (!found)
+	{
+		throw exception::EntryNotFoundException ();
+	}
+
+	KeySet ks;
+	this->_kdb.get (ks, entry.getName ());
+
+	Key k = ks.lookup (entry.getName ());
+	if (!k)
+	{
+		throw kdbrest::exception::EntryNotFoundException ();
+	}
+
+	ks.cut (entry);
+
+	if (this->_kdb.set (ks, entry.getName ()) >= 1)
+		return true;
+	else
+		return false;
+}
+
+/**
+         * @brief Can be used to determine if an entry with the given name 
+         * exists in the database or not.
+         *
+         * @param key A string containing the entry name
+         *
+         * @return true if the entry exists, false otherwise
+         */
+bool StorageEngine::entryExists (const std::string & key)
+{
+	// register read access
+	boost::shared_lock<boost::shared_mutex> lock (_mutex_entry_cache);
+
+	for (auto & elem : this->getAllEntriesRef ())
+	{
+		if (elem.getPublicName ().compare (key) == 0) return true;
+	}
+
+	return false;
+}
+
+/**
+         * @brief Can be used to get an entry from the database,
+         * base on a given name.
+         *
+         * @param key A string containing the key in elektra format
+         *
+         * @return An Entry element containing all information
+         * 
+         * @throws kdbrest::exception::EntryNotFoundException in case the
+         * requested entry could not be found
+         */
+model::Entry StorageEngine::getEntry (const std::string & key)
+{
+	// register read access
+	boost::shared_lock<boost::shared_mutex> lock (_mutex_entry_cache);
```

Comment:
I thought iterating the entries could cause trouble if another request deletes one of them at the same time. But this would change with the cachefilter plugin anyway, as soon as it works.